### ISSUE TITLE
[Bug Report] ttnn.multiply with implicit broadcast cuts off small values to 0

### ISSUE THREAD

**Issue by umalesTT** [tenstorrent/tt-metal#18955]
**Describe the bug**
When ttnn.multiply op is applied along with implicit broadcast, small values of resulting tensor are cut to 0. Because of that, the output of the code below is tensor with incorrect values ``` ttnn.Tensor([[ 0.00000,  0.00048],
             [ 0.00000,  0.00019]], shape=Shape([2, 2]), dtype=DataType::FLOAT32, layout=Layout::TILE) ```
It is clear that top left and bottom left value of this result tensor should be non-zero.

**To Reproduce**
Run:
```
import ttnn
import torch

with ttnn.manage_device(device_id=0) as device:

    a = ttnn.from_torch(torch.tensor(0.01), layout = ttnn.TILE_LAYOUT, device=device)
    b = ttnn.from_torch(torch.tensor([[0.001, 0.05],
                                      [0.004, 0.02]]), layout = ttnn.TILE_LAYOUT, device=device)
    
    res = ttnn.multiply(a, b)
    print(res)
```

**Expected behavior**
Expected output is tensor ``` ttnn.Tensor([[ 0.00001,  0.0005],
             [ 0.00004,  0.0002]], shape=Shape([2, 2]), dtype=DataType::FLOAT32, layout=Layout::TILE) ```

**Please complete the following environment information:**
- Version of software (eg. commit) https://github.com/tenstorrent/tt-metal/commit/9b3d855773

**Additional context**
This issue is relevant for any kind of model training, as it appears during optimizer pass when multiplying tensor of gradients (usually small values) with learning rate (always small value).

**Comment by wenbinlyuTT** [comment]
I have witnessed similar issues that involve implicit broadcasting & binary ops, but it's not just for small values and it's not limited to `ttnn::multiply`.
As of a709201 I can still see issues with the following reproducer on WH
```python
import torch
import ttnn

dev = ttnn.open_device(device_id=0)

# Bigger tensors fail more spectacularly, f32 fails as well
t_4d = ttnn.from_torch(torch.rand([2, 2, 1, 1]), dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=dev)
# Use 3D scalar-tensor to test implicit broadcasting
s_3d = ttnn.from_torch(torch.tensor([[[100.0]]]), dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=dev)
print("Scalar-Tensor\n", s_3d, "\nTensor 4D\n", t_4d)

print("Product\n", ttnn.multiply(t_4d, s_3d), "\nSum\n", ttnn.add(t_4d, s_3d))

# BinaryNg-based broadcasting works correctly
# print("ProductNG\n", ttnn.experimental.mul(t_4d, s_3d), "\nSumNG\n", ttnn.experimental.add(t_4d, s_3d))

ttnn.close_device(dev)
```

Output
```
Product
 ttnn.Tensor([[[[28.12500]],
               [[25.75000]]],
              [[[ 0.00000]],   <- ???
               [[ 0.00000]]]]  <- ???
Sum
 ttnn.Tensor([[[[100.50000]],
               [[100.50000]]],
               [[[ 0.44336]],  <- ???
                [[ 0.72656]]]] <- ???
```

The good news is, switching to `ttnn.experimental` solves the issue, so I guess the landing of #17160 will fix these issues.

**Comment by skhatriTT** [comment]
@cmaryanTT Do we have any update on the progress for this bug?

**Comment by KalaivaniMCW** [comment]
Hi @skhatriTT , @umalesTT ,  @cmaryanTT ,

From the pytest, I see that one input is a scalar. 
In the `eltwise/binary` (what we're going to refer to as legacy binary) infra, the Float32 support extends only to the non-broadcast case whereas in `eltwise/binary_ng` (the new binary) infra the Float32 support extends to all broadcast & non-broadcast cases. 

So for the inputs provided in the repro test, the binary operation takes place in the `BCAST` op kernel (FPU) and not the `eltwise_sfpu` kernel (SFPU) and FPU operations do no have 32-bit support (only upto 19-bit). 
In binary_ng ops this will not be a problem as we redirect to respective kernels as per the input datatype. 

If you want to retain full-precision in eltwise/binary, reshaping to have the same shapes will work. But once #16147 is merged you can turn off the `use_legacy` flag to switch to binary_ng  i.e. `ttnn.multiply(a, b, use_legacy=False)`.

<img width="1145" alt="Image" src="https://github.com/user-attachments/assets/439210c3-2280-401e-bc18-e606247460f2" />

**Comment by KalaivaniMCW** [comment]
Hi @umalesTT , @skhatriTT , 
use_legacy flag is now available.. The test passes with `ttnn.multiply(a, b, use_legacy=False)`

<img width="993" alt="Image" src="https://github.com/user-attachments/assets/be6a2ca9-1ee8-4354-bc76-c54528356041" />

**Comment by KalaivaniMCW** [comment]
Closing this issue as The test passes with ttnn.multiply(a, b, use_legacy=False) with reason https://github.com/tenstorrent/tt-metal/issues/18955#issuecomment-2788675163
